flang/include/flang/Lower/CallInterface.h

```diff
@@ -204,6 +204,11 @@ class CallInterface {
   /// Must the MLIR result be saved with a fir.save_result ?
   bool mustSaveResult() const { return saveResult; }
 
+  /// Does this interface have an implicit interface?
+  bool hasImplicitInterface() const {
```

Comment:
This name might be slightly misleading, this also returns true in context where the interface is explicit but where the function could be called without an explicit interface (see example below).
I would advise sticking with the `canBeCalledViaImplicitInterface` name here.

Example: `ifoo` interface is explicit, but `CanBeCalledViaImplicitInterface` returns true because it can be called without an explicit interface.

```
  interface
    integer function ifoo(x)
      real :: x
    end function
  end interface
  print *, ifoo(0.5)
end
```

Reply:
I'm OK with the name change, but I don't understand the example.  `ifoo()` has an explicit interface.  If I add the line
```
  print *, ifoo("abc")
```
to your example, it gets a semantic error.

Reply:
My example was not super useful indeed, what I meant is that `CanBeCalledViaImplicitInterface` returns true for `ifoo` even though it has an explicit interface. So calling the helper `hasImplicitInterface()` could be misleading because it would return true for functions that have explicit interface (but are compatible with implicit interface). I think the issue with `hasImplicitInterface` is that it could be both understood as the negation of `hasExplicitInterface` (which would be wrong here) or the fact that the interface is compatible with implicitly defined interface.